scipy/sparse/linalg/_norm.py

```diff
@@ -11,9 +11,15 @@
 
 def _sparse_frobenius_norm(x):
     if np.issubdtype(x.dtype, np.complexfloating):
-        sqnorm = abs(x).power(2).sum()
+        try:
+            sqnorm = sum(abs(x[i,j])**2 for i, j in zip(*x.nonzero()))**(1/2)
+        except TypeError:
+            sqnorm = abs(x).power(2).sum()
     else:
-        sqnorm = x.power(2).sum()
+        try:
+            sqnorm = sum(x[i,j]**2 for i, j in zip(*x.nonzero()))**(1/2)
+        except TypeError:
```

Comment:
Kindly, verify and confirm if there are more exceptions that could be thrown on failure and catch those extra exceptions